tests/ui/needless_pass_by_ref_mut.rs

```diff
@@ -270,6 +275,18 @@ pub async fn closure4(n: &mut usize) {
     })();
 }
 
+struct Data<T: ?Sized> {
+    value: T,
+}
+// Unsafe functions should not warn.
+unsafe fn get_mut_unchecked<T>(ptr: &NonNull<Data<T>>) -> &mut T {
+    &mut (*ptr.as_ptr()).value
+}
+// Unsafe blocks should not warn.
+fn get_mut_unchecked2<T>(ptr: &NonNull<Data<T>>) -> &mut T {
+    unsafe { &mut (*ptr.as_ptr()).value }
+}
+
```

Comment:
Shouldn't these be `&mut NonNull<Data<T>>`, or am I missing something?

Also, I think we could use 2 more tests here - one which ensures it doesn't lint even without any `unsafe` operations (this could be linted in the future, but being pessimistic is best) and one that ensures that, even if the prior does, rebinding it to another local (like, say, `let pdata = ptr;`) then doing smth unsafe won't lint, to show that it would track where these bindings go.

Reply:
This also creates a mutable reference from an immutable reference, that seems like UB ^^ I presume this is what the lint suggested before?